### ISSUE TITLE
[bug]: crash in v0.16.0: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by vostrnad** [lightningnetwork/lnd#7552]
### Background

```
2023-03-29 22:13:55.563 [INF] WTCL: (legacy) SessionQueue([REDACTED]) attempting to dial tower at [REDACTED]
2023-03-29 22:13:55.563 [INF] WTCL: (legacy) SessionQueue([REDACTED]) attempting to dial tower at [REDACTED]
2023-03-29 22:13:55.563 [INF] WTCL: (legacy) SessionQueue([REDACTED]) attempting to dial tower at [REDACTED]
2023-03-29 22:13:55.611 [INF] WTCL: (anchor) SessionQueue([REDACTED]) attempting to dial tower at [REDACTED]
2023-03-29 22:13:55.626 [INF] WTCL: (anchor) SessionQueue([REDACTED]) attempting to dial tower at [REDACTED]
2023-03-29 22:14:16.604 [ERR] WTCL: (legacy) SessionQueue([REDACTED]) unable to dial tower at any available Addresses: dial proxy failed: dial tcp [REDACTED]: connectex: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x209403c]

goroutine 592 [running]:
container/list.(*Element).Next(...)
        container/list/list.go:32
github.com/lightningnetwork/lnd/watchtower/wtclient.(*addressIterator).next(0xc0011a5e90, 0x0)
        github.com/lightningnetwork/lnd/watchtower/wtclient/addr_iterator.go:160 +0x9c
github.com/lightningnetwork/lnd/watchtower/wtclient.(*addressIterator).Next(0x2bb6c68?)
        github.com/lightningnetwork/lnd/watchtower/wtclient/addr_iterator.go:140 +0x1b
github.com/lightningnetwork/lnd/watchtower/wtclient.(*sessionQueue).drainBackups(0xc002a38140)
        github.com/lightningnetwork/lnd/watchtower/wtclient/session_queue.go:311 +0x210
github.com/lightningnetwork/lnd/watchtower/wtclient.(*sessionQueue).sessionManager(0xc002a38140)
        github.com/lightningnetwork/lnd/watchtower/wtclient/session_queue.go:285 +0x6f
created by github.com/lightningnetwork/lnd/watchtower/wtclient.(*sessionQueue).Start.func1
        github.com/lightningnetwork/lnd/watchtower/wtclient/session_queue.go:148 +0x5a
```

### Your environment

* lnd v0.16.0-beta
* Windows 10
* Bitcoin Core 24.0.1 as backend

**Comment by ellemouton** [comment]
thanks for reporting! will work on a fix asap. in the mean time, when you start LND, pls make sure you are adding a working tower address as it seems your LND node is unable to dial the tower.

**Comment by mans17** [comment]
Same issue here (Linux). It seems that I cannot delete a watch tower that has "unacked updates".
(I did wtclinet.enabled = false to avoid the bug.
Some threads around the web suggest to delete some files, but it seems dangerous to me :-)
e.g : https://github.com/lightningnetwork/lnd/issues/6785

**Comment by ellemouton** [comment]
just want to re-iterate that a solution for this would be to make sure a dial-able address is added for the tower in question

**Comment by ellemouton** [comment]
Hey guys - thanks again for reporting this! Closing the issue as it has now have been addressed in https://github.com/lightningnetwork/lnd/pull/7556 which will be included in `0.16.1-beta`

In the mean time, if adding a working address does not work for you (like for @mans17 ) then just deactivate the client for now. Note that future PRs will allow you to more easily remove towers :)

**Comment by mans17** [comment]
I confirm that 0.16.1-beta fixes the crash for non-responding towers, no need to deactivate the client anymore with this version.
Thank you !
